simplematrixbotlib/config.py

```diff
@@ -62,7 +62,7 @@ def join_on_invite(self, value: bool) -> None:
         self._join_on_invite = value
 
     @property
-    def allowlist(self) -> Set[str]:
+    def allowlist(self) -> Set[re.Pattern]:
```

Comment:
we could "overload" the `@property.setter` methods to also accept `re.Patterns` instead of `str` if you think users may want that for whatever reason. it might work already, didn't test (but then `_check_set_regex` needs different typing).

Reply:
`_check_set_regex` accepts `Set[str]` and returns `Set[re.Pattern]`, so either that would need to be refactored or the relevant docstrings could be fixed.